org.eclipse.xtext.xbase.tests/src/org/eclipse/xtext/xbase/tests/validation/XbaseValidationTest.xtend

```diff
@@ -1128,4 +1128,53 @@ class XbaseValidationTest extends AbstractXbaseTestCase {
 			}
 		'''.expression.assertNoErrors
 	}
+
+	@Test def void testRangeLiteralInForLoopBug440006_01() {
+		'''
+			for (int i : #[1..2]) {
+			}
+		'''.expression.assertError(XbasePackage.Literals.XBINARY_OPERATION, IssueCodes.INCOMPATIBLE_TYPES)
+	}
+
+	@Test def void testRangeLiteralInForLoopBug440006_02() {
+		'''
+			for (Integer i : #[1..2]) {
+			}
+		'''.expression.assertError(XbasePackage.Literals.XBINARY_OPERATION, IssueCodes.INCOMPATIBLE_TYPES)
+	}
+
+	@Test def void testRangeLiteralInForLoopBug440006_03() {
+		'''
+			for (Number i : #[1..2]) {
+			}
```

Comment:
This is a single for-loop iterating over something two-dimensional. I'm not in favor of supporting this because it's hiding potentially hides user's mistakes and we don't need any sugar here since it's a corner case. If somebody wants to iterate over `#[1..2]` the correct way to do so is to use two nested loops:

    for (range : #[1..2]) {
        for (Number i: range) {
            println(i)
        }
    }

It may be more illustrative when the array has one than more item:

    for (range : #[1..2, 5..9, 12..13]) {
        for (Number i: range) {
            println(i)
        }
    }